pallets/msa/src/benchmarking.rs

```diff
@@ -930,6 +930,200 @@ mod benchmarks {
+			default_name: BoundedVec::truncate_from(Vec::from("init")),
+			localized_names: BoundedBTreeMap::new(),
+			default_logo_250_100_png_cid: BoundedVec::new(),
+			localized_logo_250_100_png_cids: BoundedBTreeMap::new(),
+		};
+		// create an initial application to update
+		assert_ok!(Msa::<T>::create_application_via_governance(
+			RawOrigin::Root.into(),
+			provider_public_key.clone(),
+			initial_payload
+		));
+
+		let application_payload = ApplicationContext {
+			default_name: BoundedVec::try_from(application_name).unwrap_or_default(),
+			localized_names,
+			default_logo_250_100_png_cid: BoundedVec::try_from(cid).unwrap(),
+			localized_logo_250_100_png_cids: localized_cids,
+		};
+
+		#[extrinsic_call]
+		_(RawOrigin::Root, provider_public_key, 0u16, application_payload);
+
+		assert!(ProviderToApplicationRegistry::<T>::get(ProviderId(provider_id), 0).is_some());
+		Ok(())
+	}
+
+	#[benchmark]
+	fn propose_to_update_application() -> Result<(), BenchmarkError> {
+		let s = T::MaxProviderNameSize::get();
+		let lang_size = T::MaxLanguageCodeSize::get();
+		let _cid_size = T::MaxLogoCidSize::get();
+		let max_locale_count = T::MaxLocaleCount::get();
+
+		let application_name = (1..s as u8).collect::<Vec<_>>();
+		let cid = "bafkreidgvpkjawlxz6sffxzwgooowe5yt7i6wsyg236mfoks77nywkptdq"
+			.as_bytes()
+			.to_vec();
+		let mut localized_names = BoundedBTreeMap::new();
+		let mut localized_cids = BoundedBTreeMap::new();
+		for i in 0..max_locale_count {
+			let lang_code = make_lang_code(i as usize, lang_size as usize);
+			let lang = BoundedVec::try_from(lang_code).unwrap();
+			let name = BoundedVec::try_from(application_name.clone()).unwrap_or_default();
+			let logo = BoundedVec::try_from(cid.clone()).unwrap();
+			localized_names.try_insert(lang.clone(), name).unwrap();
+			localized_cids.try_insert(lang, logo).unwrap();
+		}
+
+		let provider_caller: T::AccountId = whitelisted_caller();
+		let (_, provider_public_key) =
+			Msa::<T>::create_account(provider_caller.clone(), EMPTY_FUNCTION).unwrap();
+		assert_ok!(Msa::<T>::create_provider_v2(
+			RawOrigin::Signed(provider_caller.clone()).into(),
+			ProviderRegistryEntry::default()
+		));
+
+		// create an initial application to update
+		let initial_payload = ApplicationContext::default();
+		assert_ok!(Msa::<T>::create_application_via_governance(
+			RawOrigin::Root.into(),
+			provider_public_key.clone(),
+			initial_payload
+		));
+
+		let application_payload = ApplicationContext {
+			default_name: BoundedVec::try_from(application_name).unwrap_or_default(),
+			localized_names,
+			default_logo_250_100_png_cid: BoundedVec::try_from(cid).unwrap(),
+			localized_logo_250_100_png_cids: localized_cids,
+		};
+
+		#[extrinsic_call]
+		_(RawOrigin::Signed(provider_public_key), 0u16, application_payload);
+
+		assert_eq!(frame_system::Pallet::<T>::events().len(), 1);
+		Ok(())
+	}
+
+	#[benchmark]
+	fn update_provider_via_governance() -> Result<(), BenchmarkError> {
```

Comment:
same as previous comment: shouldn't we parameterize this benchmark with the number of locales included in the update?